File: common/src/main/java/com/hartwig/actin/treatment/interpretation/single/OneIntegerOneString.java
package com.hartwig.actin.treatment.interpretation.single;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class OneIntegerOneString {

    public abstract int integer();

    @NotNull
    public abstract String string();
}

File: common/src/main/java/com/hartwig/actin/treatment/input/single/OneIntegerOneString.kt
package com.hartwig.actin.treatment.input.single

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class OneIntegerOneString() {
    abstract fun integer(): Int
    abstract fun string(): String
}
